### PR TITLE
userUserAgentSuffixInCosmosDiagnostics

### PR DESCRIPTION
**Pull request by xinlian12**
# Issue
If customer has configured userAgentSuffix, it is not being used in CosmosDiagnostics

# Fix
Use full user agent info in the CosmosDiagnostics
Fixed issue https://github.com/Azure/azure-sdk-for-java/issues/34857

# Examples:
userAgentSuffix: `"directClientSuffix"`
Before the change: `"userAgent":"azsdk-java-cosmos/4.45.0-beta.1 MacOSX/10.16 JRE/11.0.9"`
After the change: `"userAgent":"azsdk-java-cosmos/4.45.0-beta.1 MacOSX/10.16 JRE/11.0.9 directClientSuffix"`

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/CosmosDiagnosticsTest.java [hunk 1]
```diff
@@ -185,56 +202,44 @@ public Object[][] connectionStateListenerArgProvider() {
...
+        assertThat(diagnostics).contains("\"serializationType\":\"PARTITION_KEY_FETCH_SERIALIZATION\"");
+        assertThat(diagnostics).contains("\"userAgent\":\"" + this.gatewayClientUserAgent + "\"");
+        assertThat(diagnostics).containsAnyOf(
+            "\"machineId\":\"" + tempMachineId + "\"", // logged machineId should be static uuid or
+            "\"machineId\":\"" + ClientTelemetry.getMachineId(null) + "\"" // the vmId from Azure
+        );
+        assertThat(diagnostics).containsPattern("(?s).*?\"activityId\":\"[^\\s\"]+\".*");
+        assertThat(createResponse.getDiagnostics().getDuration()).isNotNull();
+        assertThat(createResponse.getDiagnostics().getContactedRegionNames()).isNotNull();
+        // TODO: (nakumars) - Uncomment the following line after your client telemetry fix
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/CosmosDiagnosticsTest.java:226]: uncomment or delete?

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM

**Review by kushagraThapar — APPROVED** [review]

File: sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/CosmosDiagnosticsTest.java (hunk 1 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/CosmosDiagnosticsTest.java]: Uncommented the assertions and removed the comments

**Comment by xinlian12** [comment]
Local tests for following two tests:
readMany
serializationOnVariousScenarios

**Pull request merged**